scripts/py_matter_idl/matter/idl/lint/lint_rules_parser.py

```diff
@@ -354,42 +358,6 @@ def CreateParser(file_name: str):
     """
     Generates a parser that will process a ".matter" file into a IDL
     """
-    return Parser(
-        Lark.open('lint_rules_grammar.lark', rel_to=__file__, parser='lalr', propagate_positions=True, maybe_placeholders=True), file_name=file_name)
-
-
-if __name__ == '__main__':
-    # This Parser is generally not intended to be run as a stand-alone binary.
-    # The ability to run is for debug and to print out the parsed AST.
-    import click
-
-    # Supported log levels, mapping string values required for argument
-    # parsing into logging constants
-    __LOG_LEVELS__ = {
-        'debug': logging.DEBUG,
-        'info': logging.INFO,
-        'warn': logging.WARN,
-        'fatal': logging.FATAL,
-    }
```

Comment:
You don't have to hardcode logging log levels, you can use:

```
>>> __LOG_LEVELS__ = { name.lower(): value for (name,value) in logging.getLevelNamesMapping().items() if value > 0 }
>>> __LOG_LEVELS__
{'critical': 50, 'fatal': 50, 'error': 40, 'warn': 30, 'warning': 30, 'info': 20, 'debug': 10}
>>> 
```

This is available from Python 3.11 so the application depends which Python versions we support in the SDK. Python 3.8 is used in ubuntu-18.04 for example (in extended support until 2028).